### PR TITLE
Notify Sentry About Jobs With an Internal Error

### PR DESCRIPTION
**Pull request by ghickman**
This reports jobs with the status message "Internal error" to Sentry as message events (instead of exceptions) so we can start getting notifications about them.

### CODE REVIEW

**Review by evansd — APPROVED** [review]
Perfect! Thanks for this.

We can do something more sophisticated than `if "internal error"` once we make better use of the "status_codes" system but this will do the trick for now.

File: jobserver/api/jobs.py [hunk 1]
```diff
@@ -119,6 +119,16 @@ def post(self, request, *args, **kwargs):
                 else:
                     updated_job_ids.append(str(job.id))
 
+                if "internal error" in job.status_message.lower():
```
Comment (benbc) [inline jobserver/api/jobs.py:122]: Am I right that this string is generated in job-runner [here](https://github.com/opensafely-core/job-runner/search?q=%22internal+error%22+filename%3Arun.py)?

This seems like a good immediate solution to a current problem, but I guess that ideally we'd want something a bit more structured so we're not depending on string matching?

File: tests/unit/jobserver/api/test_jobs.py [hunk 2]
```diff
@@ -442,6 +442,37 @@ def test_jobapiupdate_post_only(api_rf):
...
+            "updated_at": now,
+            "completed_at": None,
+        },
+    ]
+    request = api_rf.post(
+        "/", HTTP_AUTHORIZATION=backend.auth_token, data=data, format="json"
+    )
+    response = JobAPIUpdate.as_view()(request)
+
+    assert response.status_code == 200, response.data
```
Comment (benbc) [inline tests/unit/jobserver/api/test_jobs.py:473]: Is it possible to check that that we send the error to Sentry?

File: jobserver/api/jobs.py (hunk 1 above)
Comment (benbc) [reply jobserver/api/jobs.py:122]: I've just seen Dave's comment. :+1:

Comment (evansd) [reply jobserver/api/jobs.py:122]: @benbc We already have the basic building blocks to do this, but just aren't actually using them. Jobs have a `status_code` field whose values are defined [here](https://github.com/opensafely-core/job-runner/blob/4ef28a9c2e64d523439536c21663447c3f28bd86/jobrunner/models.py#L26-L36) and which is pushed up to job-server along with everything else. We need to add an `internal_error` code and then set that in the appropriate place.

Comment (benbc) [reply jobserver/api/jobs.py:122]: Great.

File: tests/unit/jobserver/api/test_jobs.py (hunk 2 above)
Comment (ghickman) [reply tests/unit/jobserver/api/test_jobs.py:473]: Good catch, I've added an assert to check we're calling `capture_message()`.  It doesn't look like sentry ships any of their testing tools with the SDK for checking you've called it correctly unfortunately, would be nicer than mocking.

**Pull request merged**